src/core/filesystem.cc

```diff
@@ -634,8 +634,8 @@ S3FileSystem::ParsePath(
 S3FileSystem::S3FileSystem(
     const Aws::SDKOptions& options, const std::string& s3_path)
     : options_(options), s3_regex_(
-                             "s3://([0-9a-zA-Z-.]+):([0-9]+)/([0-9a-z.-]+)(((/"
-                             "[0-9a-zA-Z.-_]+)*)?)")
+                             "s3://([0-9a-zA-Z.-]+):([0-9]+)/([0-9a-z.-]+)(((/"
```

Comment:
Why does just moving it in the group change anything. If it is a special character shouldn't wee escape it somehow?

Reply:
As described by this [comment](https://github.com/NVIDIA/triton-inference-server/issues/1534#issuecomment-632499547).

`.-_` imply range between `.` to `_` , `._-` will match character between `.`, `-`, `_`.

This is because `-` also acts as a range char in regex. Yes I believe since it is a special char escaping it would be the appropriate way to go about it. Fixing it now.